data-prepper-api/src/main/java/org/opensearch/dataprepper/model/annotations/DataPrepperExtensionPlugin.java

```diff
@@ -13,11 +13,18 @@
 @Retention(RetentionPolicy.RUNTIME)
 @Target({ElementType.TYPE})
 public @interface DataPrepperExtensionPlugin {
+    String DEFAULT_DEPRECATED_ROOT_KEY_JSON_PATH = "";
+
     /**
      * @return extension plugin configuration class.
      */
     Class<?> modelType();
 
+    /**
+     * @return valid JSON path string starts with "/" pointing towards the configuration block.
+     */
+    String deprecatedRootKeyJsonPath() default DEFAULT_DEPRECATED_ROOT_KEY_JSON_PATH;
```

Comment:
Should this be a collection to allow for multiple deprecated values?

Reply:
I agree it could be a collection but for now I did not see we would change the name often. So I tend to keep it consistent with the deprecatedName in `DataPrepperPlugin`.